source/extensions/filters/http/kill_request/kill_request_config.cc

```diff
@@ -0,0 +1,32 @@
+#include "extensions/filters/http/kill_request/kill_request_config.h"
+
+#include "envoy/extensions/filters/http/kill_request/v3/kill_request.pb.h"
+#include "envoy/extensions/filters/http/kill_request/v3/kill_request.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+#include "extensions/filters/http/kill_request/kill_request_filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace KillRequest {
+
+Http::FilterFactoryCb KillRequestFilterFactory::createFilterFactoryFromProtoTyped(
+    const envoy::extensions::filters::http::kill_request::v3::KillRequest& proto_config,
+    const std::string&, Server::Configuration::FactoryContext& context) {
+  return [proto_config, &context](Http::FilterChainFactoryCallbacks& callbacks) -> void {
+    callbacks.addStreamFilter(
+        std::make_shared<KillRequestFilter>(proto_config, context.api().randomGenerator()));
+  };
+}
+
+/**
+ * Static registration for the KillRequest filter. @see RegisterFactory.
+ */
+REGISTER_FACTORY(KillRequestFilterFactory,
+                 Server::Configuration::NamedHttpFilterConfigFactory){"envoy.kill_request"};
```

Comment:
See comment below about deprecated names. In fact, all extension names are now effectively deprecated. @kyessenov should we have some way to register without declaring these at all?

Reply:
Yes, we can just do `REGISTER_FACTORY(KillRequestFilterFactory, Server::Configuration::NamedHttpFilterConfigFactory)`
https://github.com/envoyproxy/envoy/blob/9ccd778d4d7ccd986419620ed2e93b9796116863/include/envoy/registry/registry.h#L504

Reply:
Yeah, the name argument here is supposed to be an alias I think. Factories have unique names with aliases.